### PR TITLE
chore: create a npm-publish github actions workflow

### LINKED ISSUE #4: Add npm build
Build npm package  and publish it

### PR DESCRIPTION
**Pull request by mfklauberg**
This PR introduces a workflow which aims to publish correctly the library code when a new [GitHub release](https://github.com/philips-software/odin/releases) is created.

The workflow is composed by two separate jobs, `build` and `publish-npm`. The `build` job will build, which also runs the linter in the code, and run the project test suite. After, the `publish-npm` job will run, with the main responsibility of publish the project to the [NPM registry](https://www.npmjs.com) under the [@odinjs/odin](https://www.npmjs.com/package/@odinjs/odin) scope.

If there's any more necessary steps, please let me know.

closes #4

### CODE REVIEW

File: .github/workflows/npm-publish.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        with:
+          node-version: 14
+          registry: https://registry.npmjs.org/
+          scope: '@odinjs'
+      - name: Install dependencies
+        run: yarn
+      - name: Build
+        run: yarn build
+      - name: Publish
+        run: yarn publish --access publich
```
Comment (herberttn) [inline .github/workflows/npm-publish.yml:43]:
```suggestion
        run: yarn publish --access public
```

File: .github/workflows/npm-publish.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# This workflow is responsible for building, testing and publishing the library when a new release tag is created.
+# The workflow is composed by two jobs, `build` and `publish-npm`. The `build` job will build and test the code, and then,
+# if everything is working correctly, it will try and publish the code the
```
Comment (herberttn) [inline .github/workflows/npm-publish.yml:3]: Incomplete sentence

**Review by herberttn — CHANGES_REQUESTED** [review]
I think it would be good to also change the node version on `.travis.yml`, so that the pull request checks use a matching version.

**Review by herberttn — APPROVED** [review]

**Review by JeroenKnoops — APPROVED** [review]

File: .travis.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@ dist: trusty
 language: node_js
 cache: yarn
 node_js:
-- 8
+- 14
```
Comment (herberttn) [inline .travis.yml:5]: Actually I think we can delete the Travis file, since we'll now be using Github Actions.

**Review by herberttn — CHANGES_REQUESTED** [review]

**Comment by JeroenKnoops** [comment]
I've added the NPM_TOKEN in the secrets.. This will be published with user: philips-software-fa

**Review by herberttn — APPROVED** [review]

**Comment by herberttn** [comment]
> I've added the NPM_TOKEN in the secrets.. This will be published with user: philips-software-fa

Thank you @JeroenKnoops!

**Pull request merged**